test/e2e/field_selectors/field_selectors_test.go

```diff
@@ -0,0 +1,196 @@
+package field_selectors_test
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	"github.com/flightctl/flightctl/test/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	deviceYAMLPath = "examples/device.yaml"
+	fleetYAMLPath  = "examples/fleet.yaml"
+)
+
+// Define the struct for parsing device.yaml
+type Device struct {
+	Metadata struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestFieldSelectors(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Field Selectors E2E Suite")
+}
+
+var _ = Describe("Field Selectors in Flight Control", func() {
+
+	var (
+		harness *e2e.Harness
+		token   string
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+
+		loginArgs := []string{"login", "${API_ENDPOINT}", "--insecure-skip-tls-verify"}
+		if token != "" {
+			loginArgs = append(loginArgs, "--token", token)
+		}
+
+		out, err := harness.CLI(loginArgs...)
+		if strings.Contains(out, "You must obtain an API token by visiting") {
+			token, err = harness.SH("oc", "whoami", "-t")
+			token = strings.TrimSpace(token)
+			Expect(err).ToNot(HaveOccurred(), "Failed to retrieve token")
+			Expect(token).ToNot(BeEmpty(), "Token from 'oc whoami' should not be empty")
+
+			loginArgs = append(loginArgs, "--token", token)
+			out, err = harness.CLI(loginArgs...)
+		}
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(strings.TrimSpace(out)).To(BeElementOf("Auth is disabled", "Login successful"))
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(false)
+		_, _ = harness.CLI("delete", "fleet")
+	})
+
+	// Helper function to dynamically extract device name
+	extractDeviceName := func() string {
+		deviceName, err := extractNameFromYAML(filepath.Join(util.GetTopLevelDir(), deviceYAMLPath))
+		Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to extract name from %s", deviceYAMLPath))
+		return strings.TrimSpace(deviceName)
+	}
+
+	Context("login", func() {
+		It("should have worked, and we can list devices", func() {
+			out, err := harness.CLI("get", "devices")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(ContainSubstring("NAME"))
+		})
+	})
+
+	Context("apply/recursive", func() {
+		It("should work for a complete set of yamls", func() {
+			deviceName := extractDeviceName()
+
+			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), deviceYAMLPath))
```

Comment:
```suggestion
			out, err := harness.CLI("apply", "-R", "-f", filepath.Join(util.GetTopLevelDir(), "examples"))
```
Did you mean to apply all the examples here?

Reply:
but you apply only one yaml here with deviceYAMLPath

Reply:
you apply only device here